simple/msg_sockets.c

```diff
@@ -334,7 +335,8 @@ static int setup_handle(void)
 		FT_PRINTERR("fi_getinfo", ret);
 		goto out;
 	}
-	free(fi_pep->src_addr);
+	/* save source address pointer for later restore */
+	saved_src_addr = fi_pep->src_addr;
```

Comment:
Can this be saved/restored in the same function?

It's not great that the app is treating fi_info in this way, but what is windows doing that breaks this?